PapiInstance.cpp

```diff
@@ -1,137 +1,331 @@
 #include "PapiInstance.h"
 #include <algorithm>
+#include <vector>
+#include <string.h>
```

Comment:
nitpick: Please remove the `.h` part in the include, as it is not required for C++.